concepts/pointers/about.md

```diff
@@ -0,0 +1,404 @@
+
+A type followed by a `*` followed by an identifier is the way to declare a pointer to a variable of that type.
+
+Okay, but why make a variable just to hold the address of another variable? 
+
+Why do we care what the address of a variable is in the first place?
+
+Well, we care where variables are when they are in an array.
+
+## Arrays and Pointers
+
+When we create an array like so
+
+```c
+int my_ints[] = {1, 2};
+```
+
+the value of `my_ints` is actually its address, which also happens to be the same as the address of its first element. The address of its second element will be the first element's address plus the size of an int on the system.
+
+To better see this, try running the following
+
+```c
+#include <stdio.h>
+
+int main() {
+    // my_int's value is its address
+    int my_ints[] = {1, 2};
+    
+    // since my_ints is already an address
+    // we don't need the address operator to make a pointer from it.
+    // a pointer to my_ints holds the address of my_ints
+    int *my_ptr = my_ints;
+    
+    // the address of my_ints and the value of my_ptr are the same
+    printf("my_ints is %p. my_ptr is %p.\n", my_ints, my_ptr);
+    
+    // both addresses of the first element are the same as the original addresses
+    printf("&my_ints[0] is %p. my_ptr + 0 is %p.\n", &my_ints[0], my_ptr + 0);
+    
+    // both addresses for the second element have increased by the size of an int
+    printf("&my_ints[1] is %p. my_ptr + 1 is %p.\n", &my_ints[1], my_ptr + 1); // 
+    
+    return 0;
+}
+```
+
+## Pointer Arithmetic, or Why Array Indexes begin at 0
+
+The previous example shows why array indexing begins at `0` instead of `1`.
+The index into an array multiplies the index by the size of the type and adds it to the starting address of the array.
+Since the first element aligns with the beginning of the array, we multiply the type by zero for adding to the start address to get the first element.
+To get the second element we multiple the type by 1 for adding to the start address.
+If the type is four bytes in length, the second element will be four bytes higher than the first element.
+
+This is also why pointers are declared to be of a certain type.
+If a pointer did not know what type it points to, it would not know how many bytes to advance when adding `1` to it.
+Adding `1` to a pointer does not necessariy mean we're adding one byte to the address.
+It means we're adding `1` times the length of its type in bytes.
+
+But wait, there's more! You didn't think we were finished with `*`, did you?
+
+## Dereference Operator
+
+Now that we understand how addresses work with array indexing, what good is it if we don't get the value stored at an address?
+How do we do that?
+To get the value stored at an address we use the dereference operator, which is, you guessed it, `*`.
+
+Come on!
+Are you kidding me?
+How many uses are there for `*`?!
+Multiplication, declaring a pointer, and now derefencing!
+It's confusing!
```

Comment:
```suggestion
It's confusing!

Yes. Yes it is.
```
🤣 😄

Reply:
Well, I at least fixed the spelling for "dereferencing." 😄

Reply:
"Derefencing": The taking down of a refurbished fence. 😄